crates/relayer/src/chain/cosmos/gas.rs

```diff
@@ -25,6 +29,21 @@ pub fn gas_amount_to_fee(config: &GasConfig, gas_amount: u64) -> Fee {
     }
 }
 
+pub async fn dynamic_gas_price(config: &GasConfig, rpc_address: &Url) -> GasPrice {
+    if let Some(dynamic_gas_price_multiplier) = config.dynamic_gas_price_multiplier {
+        query_eip_base_fee(&rpc_address.to_string())
+            .await
+            .map(|base_fee| base_fee * dynamic_gas_price_multiplier)
+            .map(|new_price| GasPrice {
+                price: new_price,
+                denom: config.gas_price.denom.clone(),
+            })
+            .unwrap_or_else(|_| config.gas_price.clone())
```

Comment:
I'm wondering if we should at least output a warning with the error if the `query_eip_base_fee` fails here. What do you think?

Reply:
Yeah good idea!